### PR TITLE
reference/garbage-collection: Add a table to explain GC's configurations about mode and concurrency

### PR DESCRIPTION
**Pull request by MyonKeminta**
### What is changed, added or deleted? 

I found that it's still difficult to understand the difference of GC between 2.x and 3.0. This PR tries to describe it in a more understandable way by adding a table for some of the configurations.

### What is the related PR or file link(s)? 

N/A

### Which version does your change affect? 

v3.0 & dev

### CODE REVIEW

**Comment by MyonKeminta** [comment]
@zhangjinpeng1987 PTAL if you have time. Thanks!

File: dev/reference/garbage-collection/configuration.md [hunk 1]
```diff
@@ -87,3 +87,22 @@ update mysql.tidb set VARIABLE_VALUE="24h" where VARIABLE_NAME="tikv_gc_life_tim
 
 - 手动设置 GC concurrency。要使用该参数，必须将 [`tikv_gc_auto_concurrency`](#tikv-gc-auto-concurrency) 设为 `false` 。
 - 默认值：2
+
+## 关于 GC 流程的说明
+
+从 3.0 起，由于分布式 GC 模式和并行 Resolve Locks 的支持，部分配置选项的作用发生了变化。可根据下表理解不同版本中这些配置的区别：
```
Comment (yikeke) [inline dev/reference/garbage-collection/configuration.md]:
```suggestion
从 TiDB 3.0 版本起，由于对分布式 GC 模式和并行 Resolve Locks 的支持，部分配置选项的作用发生了变化。可根据下表理解不同版本中这些配置的区别：
```

File: dev/reference/garbage-collection/configuration.md [hunk 2]
```diff
@@ -87,3 +87,22 @@ update mysql.tidb set VARIABLE_VALUE="24h" where VARIABLE_NAME="tikv_gc_life_tim
...
+
+| 版本/配置          |  Resolve Locks          |  Do GC  |
```
Comment (yikeke) [inline dev/reference/garbage-collection/configuration.md]:
```suggestion
| 版本/配置          |  Resolve Locks          |  执行 GC  |
```

Comment (MyonKeminta) [reply dev/reference/garbage-collection/configuration.md]: This is consistent with the name of the third phase of GC, which is documented here: https://pingcap.com/docs-cn/v3.0/reference/garbage-collection/overview/

Comment (yikeke) [reply dev/reference/garbage-collection/configuration.md]: Got it.

**Comment by MyonKeminta** [comment]
Sorry I just noticed that I forgot to address comments in 3.0 doc

**Comment by MyonKeminta** [comment]
ping @dcalvin @yikeke

**Comment by MyonKeminta** [comment]
ping @dcalvin @yikeke

File: dev/reference/garbage-collection/configuration.md [hunk 3]
```diff
@@ -94,3 +94,22 @@ update mysql.tidb set VARIABLE_VALUE="24h" where VARIABLE_NAME="tikv_gc_life_tim
...
+|-------------------|---------------|----------------|
+| 2.x               | 串行 | 并行 |
+| 3.0 <br/> `tikv_gc_mode = centered` <br/> `tikv_gc_auto_concurrency = false` | 并行 | 并行 |
+| 3.0 <br/> `tikv_gc_mode = centered` <br/> `tikv_gc_auto_concurrency = true` | 自动线程数并行 | 自动线程数并行 |
+| 3.0 <br/> `tikv_gc_mode = distributed` <br/> `tikv_gc_auto_concurrency = false` | 并行 | 分布式 |
+| 3.0 <br/> `tikv_gc_mode = distributed` <br/> `tikv_gc_auto_concurrency = true` <br/> （默认配置） | 自动线程数并行 | 分布式 |
+
+解释：
+
+- 串行：由 TiDB 逐个 Region 发送请求。
```
Comment (yikeke) [inline dev/reference/garbage-collection/configuration.md]:
```suggestion
- 串行：由 TiDB 逐个向 Region 发送请求。
```

File: v3.0/reference/garbage-collection/configuration.md [hunk 4]
```diff
@@ -94,3 +94,22 @@ update mysql.tidb set VARIABLE_VALUE="24h" where VARIABLE_NAME="tikv_gc_life_tim
...
+|-------------------|---------------|----------------|
+| 2.x               | 串行 | 并行 |
+| 3.0 <br/> `tikv_gc_mode = centered` <br/> `tikv_gc_auto_concurrency = false` | 并行 | 并行 |
+| 3.0 <br/> `tikv_gc_mode = centered` <br/> `tikv_gc_auto_concurrency = true` | 自动线程数并行 | 自动线程数并行 |
+| 3.0 <br/> `tikv_gc_mode = distributed` <br/> `tikv_gc_auto_concurrency = false` | 并行 | 分布式 |
+| 3.0 <br/> `tikv_gc_mode = distributed` <br/> `tikv_gc_auto_concurrency = true` <br/> （默认配置） | 自动线程数并行 | 分布式 |
+
+解释：
+
+- 串行：由 TiDB 逐个 Region 发送请求。
```
Comment (yikeke) [inline v3.0/reference/garbage-collection/configuration.md]:
```suggestion
- 串行：由 TiDB 逐个向 Region 发送请求。
```

File: dev/reference/garbage-collection/configuration.md [hunk 5]
```diff
@@ -94,3 +94,22 @@ update mysql.tidb set VARIABLE_VALUE="24h" where VARIABLE_NAME="tikv_gc_life_tim
...
+
+| 版本/配置          |  Resolve Locks          |  Do GC  |
+|-------------------|---------------|----------------|
+| 2.x               | 串行 | 并行 |
+| 3.0 <br/> `tikv_gc_mode = centered` <br/> `tikv_gc_auto_concurrency = false` | 并行 | 并行 |
+| 3.0 <br/> `tikv_gc_mode = centered` <br/> `tikv_gc_auto_concurrency = true` | 自动线程数并行 | 自动线程数并行 |
+| 3.0 <br/> `tikv_gc_mode = distributed` <br/> `tikv_gc_auto_concurrency = false` | 并行 | 分布式 |
+| 3.0 <br/> `tikv_gc_mode = distributed` <br/> `tikv_gc_auto_concurrency = true` <br/> （默认配置） | 自动线程数并行 | 分布式 |
+
+解释：
```
Comment (AndreMouche) [inline dev/reference/garbage-collection/configuration.md]: 关键字说明

File: dev/reference/garbage-collection/configuration.md [hunk 6]
```diff
@@ -94,3 +94,22 @@ update mysql.tidb set VARIABLE_VALUE="24h" where VARIABLE_NAME="tikv_gc_life_tim
...
+
+- 串行：由 TiDB 逐个向 Region 发送请求。
+- 并行：使用 `tikv_gc_concurrency` 选项所指定的线程数，并行地向每个 Region 发送请求。
+- 自动线程数并行：使用 TiKV 节点的个数作为线程数，并行地向每个 Region 发送请求。
```
Comment (AndreMouche) [inline dev/reference/garbage-collection/configuration.md]: 默认并行

Comment (yikeke) [reply dev/reference/garbage-collection/configuration.md]: PTAL again. @AndreMouche

**Pull request merged**